kernel/sched.c

```diff
@@ -959,6 +959,9 @@ void *z_get_next_switch_handle(void *interrupted)
 			 * will not return into it.
 			 */
 			if (z_is_thread_queued(old_thread)) {
+#ifdef CONFIG_SCHED_IPI_CASCADE
+				flag_ipi(ipi_mask_create(old_thread));
```

Comment:
Not normally a fan of premature optimization, but this is a circumstance where it might be worthwhile, given the O(N) loop in ipi_mask_create().

* You don't need to do this if the incoming thread is runnable on all CPUs (i.e. has a mask of -1, the default/common case), because that's isomorphic to normal scheduling, right?
* You don't need to do this if the outgoing thread is limited to the current CPU, which is another very common case.
* In particular, if CPU_MASK_PIN_ONLY=y, you know you never need to do this at all.